index.js

```diff
@@ -277,11 +284,17 @@ function onReady () {
             elements[json.targetID].unmaximize()
             break;
             case consts.eventNames.windowCmdUpdateCustomOptions:
-            windowOptions[json.targetID] = json.windowOptions
-            client.write(json.targetID, consts.eventNames.windowEventUpdatedCustomOptions, json.windowOptions)
+            client.write(json.targetID, consts.eventNames.windowEventUpdatedCustomOptions, json.windowOptions);
             break;
             case consts.eventNames.windowCmdWebContentsExecuteJavascript:
-            elements[json.targetID].webContents.executeJavaScript(json.code).then(() => client.write(json.targetID, consts.eventNames.windowEventWebContentsExecutedJavaScript));
+            elements[json.targetID].webContents.executeJavaScript(json.code)
+                .then(() => client.write(json.targetID, consts.eventNames.windowEventWebContentsExecutedJavaScript));
+            break;
+            case consts.eventNames.windowCmdWebContentsSetProxy:
+            elements[json.targetID].webContents.session.clearAuthCache().then(() => {
```

Comment:
I don't really like the fact of clearing the Session Auth Cache here.

It should be split into 2 different events, one for clearing the auth cache and another one to set the proxy.

Reply:
I disagree since when you re-set the proxy you don't want it to try use the same credentials (for instance, when using a rotating proxy IP), the behaviour in Electron is what is strange.

Reply:
Thing is, it makes sense in your case, but not in others. Clearing auth cache shouldn't be binded to setting the proxy simply to allow developers to do one without the other.